### DISCUSSION TITLE
Integrating DeepSeek-Coder with Zed IDE - Assistance Needed

### DISCUSSION THREAD

**Discussion by erik-balfe** [zed-industries/zed#14199] [Feature Requests]
Hello Zed community,

I'm trying to integrate the DeepSeek Coder language model into Zed IDE as an AI assistant, but I'm encountering some issues. I'd appreciate any help or guidance from those who might have experience with this integration.

Current setup:
I've added the following configuration to my settings:

```json
"assistant": {
  "defaultopenaimodel": "deepseek-coder",
  "version": "1",
  "provider": {
    "name": "deepseek",
    "type": "openai",
    "defaultmodel": "deepseek-coder",
    "apiurl": "https://api.deepseek.com/v1"
  }
}
```

Issue:
Despite DeepSeek stating that their API is compatible with OpenAI's, I haven't been able to get it working. I've tried both with and without the "v1" in the `apiurl, but there's no progress. The assistant panel still shows only GPT-4 and GPT-3.5 models in the dropdown menu.

Questions:
1. Has anyone successfully integrated  DeepSeek-coder or any other openai compatible ones with Zed IDE?
2. Are there any specific steps or additional configuration needed for this integration?
3. Is there a way to debug why the another model isn't appearing in the assistant panel?

Any insights, suggestions, or documentation links would be greatly appreciated. Thank you in advance for your help!

Environment details:
- Zed IDE version: 0.143.6
- Operating System: Fedora 40

**Comment by clyer** [comment]
https://zed.dev/releases/stable/0.142.4
```json
"provider": {
      "name": "openai",
      "type": "openai",
      "default_model": {
        "custom": {
          "name": "deepseek-coder",
          "max_tokens": 32000
        }
      },
      "available_models": [
        {
          "custom": {
            "name": "deepseek-chat",
            "max_tokens": 32000
          }
        },
        {
          "custom": {
            "name": "deepseek-coder",
            "max_tokens": 32000
          }
        }
      ],
      "api_url": "https://api.deepseek.com/v1"
    }
```

**Reply by nedzen** [reply]
How should I change this code if I want to run the model locally ?

**Reply by KRollenbeck** [reply]
You could try running it throug LM Studio or Ollama. Or you could change the api_url to your localhost.

**Comment by erik-balfe** [comment]
Also `max_tokens` should be set to 128000 as both models support this max context window.

**Comment by zhaozhao** [comment]
## 🎉🎉 Deepseek is now offically supported by Zed with version 0.170.4

 Fill in the deepseek api key in the AI assistent configure panel, every things is set to go!

 You don't need any extra configure from now on.

BTW, you may want to try silliconflow as deepseek provider which offers 20 million tokens for free
give it a try: https://cloud.siliconflow.cn/i/biWwZgHI

----
For newer zed version 0.149.5, this setting works for me:
```
  {
    "language_models": {
      "openai": {
        "version": "1",
        "available_models": [
          {
            "name": "deepseek-coder",
            "max_tokens": 128000
          },
          {
            "name": "deepseek-chat",
            "max_tokens": 128000
          }
        ],
        "api_url": "https://api.deepseek.com/beta"
      }
    },
    "assistant": {
      "default_model": {
        "provider": "openai",
        "model": "deepseek-chat"
      },
      "version": "2"
    }
  }
```
Do not forget to set api_key of deepseek in zed assistant configure page.

**Reply by zhaozhao** [reply]
you may want to try silliconflow as deepseek provider which offers 20 million tokens for free
give it a try: https://cloud.siliconflow.cn/i/biWwZgHI

**Reply by dreamerlzl** [reply]
Also remember to update your OPENAI_API_KEY environment variable, haha

**Comment by d2luu** [comment]
does anyone know how to config temperature of deepseek-coder in zed?

**Reply by erik-balfe** [reply]
I couldnt find any mentions of temperature, so I suppose its currently not implemented to adjust that setting for LMs. But I've never faced any problems with temperature with deepseek. You may try to fix your problem with setting custom system prompt as a temporary solution.

**Comment by carlz32** [comment]
Has anyone encountered this error `Matches multiple schemas when only one must validate. ` when config your zed `assistant`?
<img width="1031" alt="image" src="https://github.com/user-attachments/assets/6a9e117f-aa7d-4c72-8609-7498de75d020" />

**Reply by zakke27** [reply]
same warning, maybe this is a problem with the settings json schema.

**Comment by gmdias727** [comment]
zed settings.json

```json
{
  "assistant": {
    "default_model": {
      "provider": "copilot_chat",
      "model": "gpt-4o"
    },
    "version": "2"
  },
  "provider": {
    "name": "openai",
    "type": "openai",
    "api_key": "<key>",
    "api_url": "https://api.openai.com/v1/engines",
    "default_model": {
      "custom": {
        "name": "deepseek-coder",
        "max_tokens": 128000
      }
    },
    "available_models": [
      {
        "custom": {
          "name": "deepseek-chat",
          "max_tokens": 128000
        }
      },
      {
        "custom": {
          "name": "deepseek-coder",
          "max_tokens": 128000
        }
      }
    ],
  }
}
```
What am I missing here in order to add DeepSeek as an LMM into Zed?

![image](https://github.com/user-attachments/assets/77f6b4ab-b744-454c-abcc-c2bbfe91e5a5)

Only shows those LLM's at this moment.

**Reply by carlz32** [reply]
try @zhaozhao 's config, that works for me in the latest version.

**Reply by GregFedy** [reply]
Hi, how to integrate deepseek api in zed? https://platform.deepseek.com/usage
